agent/exploits/cve_2023_49897.py

```diff
@@ -0,0 +1,117 @@
+"""Agent Asteroid implementation for CVE-2023-49897."""
+import re
+
+import requests
+from ostorlab.agent.kb import kb
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+from requests import exceptions as requests_exceptions
+
+from agent import definitions
+from agent import exploits_registry
+
+VULNERABILITY_TITLE = 'FXC wireless LAN routers "AE1021PE" and "AE1021" vulnerable to OS command injection.'
+VULNERABILITY_REFERENCE = "CVE-2023-49897"
+VULNERABILITY_DESCRIPTION = (
+    "An OS command injection vulnerability exists in AE1021PE firmware version 2.0.9 and earlier "
+    "and AE1021 firmware version 2.0.9 and earlier. If this vulnerability is exploited, "
+    "an arbitrary OS command may be executed by an attacker who can log in to the product."
+)
+
+DEFAULT_TIMEOUT = 90
+MAX_REDIRECTS = 2
+HEADERS = {
+    "Content-Type": "application/x-www-form-urlencoded",
+    "Cookie": "cookieno=489646; username=admin; password=admin",
+}
+
+
+@exploits_registry.register
+class CVE202349897Exploit(definitions.Exploit):
+    """
+    CVE-2023-49897: FXC wireless LAN routers \"AE1021PE\" and \"AE1021\" vulnerable to OS command injection.
+    """
+
+    def __init__(self) -> None:
+        self.session = requests.Session()
+        self.session.max_redirects = MAX_REDIRECTS
+        self.session.verify = False
+
+    def accept(self, target: definitions.Target) -> bool:
+        target_uri = f"{target.scheme}://{target.host}:{target.port}"
+        data = {
+            "username_input": "admin",
+            "password_input": "admin",
+            "lang": "ja_JP",
+            "hashstr": "202310281340",
+            "username": "admin",
+            "password": "admin",
+        }
+        try:
+            self.session.post(
+                target_uri + "/cgi-bin/login.apply",
+                headers=HEADERS,
+                data=data,
+                timeout=DEFAULT_TIMEOUT,
+            )
+            response = self.session.get(
+                target_uri + "/main_top.htm", headers=HEADERS, timeout=DEFAULT_TIMEOUT
+            )
+            return response.status_code == 200 and "logout" in response.text
+        except requests_exceptions.RequestException:
+            return False
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        target_uri = f"{target.scheme}://{target.host}:{target.port}"
+        try:
+            response = self.session.get(
+                target_uri + "/cgi-bin/runtime?system_status",
+                headers=HEADERS,
+                timeout=DEFAULT_TIMEOUT,
+            )
+            if response.status_code == 200:
+                pattern = re.compile(r"firmware_version=(\d+\.\d+\.\d+)")
+                match = pattern.search(response.text)
+                if match is not None:
+                    firmware_version = match.group(1)
+                    if firmware_version <= "2.0.9":
```

Comment:
"2.0.10" <= "2.0.9" would return `True`

Use `packaging` to compare versions instead